frontend/src/components/eMIB/Emib.jsx

```diff
@@ -53,12 +78,94 @@ class Emib extends Component {
     }
   };
 
-  openPopup = () => {
-    this.setState({ showPopupBox: true });
+  openSubmitPopup = () => {
+    this.setState({ showSubmitPopup: true });
   };
 
   closePopup = () => {
-    this.setState({ showPopupBox: false });
+    this.setState({ showSubmitPopup: false, showQuitPopup: false });
+    //reset all checkbox states on close
+    this.resetCheckboxStates();
+  };
+
+  openQuitPopup = () => {
+    this.setState({ showQuitPopup: true });
+  };
+
+  firstCheckboxToggle = () => {
```

Comment:
I think the logic makes sense here, but there's quite a bit of duplicated code. Let's pair on this and see if we can come up with something a little lighter?

Reply:
We did some pair programming to simplified that. See last 6 commits.